### PR TITLE
first version of websocket client

### PR DESCRIPTION
**Pull request by gallison93**

### CODE REVIEW

File: src/mtev_websocket_client.c [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+#ifdef HAVE_WSLAY
+  int fd = -1, rv;
+  struct addrinfo *res = NULL, *res_ptr = NULL, hints;
+  char port_str[6];
+
+  snprintf(port_str, 6, "%d", port);
+  memset(&hints, 0, sizeof(hints));
+  hints.ai_family = AF_UNSPEC;
+  hints.ai_socktype = SOCK_STREAM;
+  if((rv = getaddrinfo(host, port_str, &hints, &res))) {
```
Comment (postwait) [inline src/mtev_websocket_client.c]: This is a blocking call. I would have this function just take an IPv4 or IPv6 address -- no lookup.  The caller can struggle with the non-blocking lookup.

File: src/mtev_websocket_client.c [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+      client->sent_handshake = mtev_true;
+    }
+    
+    if(client->sent_handshake && mask & EVENTER_READ) {
+      if(mtev_websocket_client_recv_handshake(client) == mtev_false) {
+        mtevL(mtev_error, "mtev_websocket_client_recv_handshake failed, aborting drive\n");
+        goto abort_drive;
+      }
+      wslay_event_context_client_init(&client->wslay_ctx, &wslay_callbacks, client);
+      client->did_handshake = mtev_true;
```
Comment (postwait) [inline src/mtev_websocket_client.c]: I would add a client call back here (and in the case) so the programmer can react to the completion "ready state" of the websockets connection.

File: src/mtev_websocket_client.c [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+    eventer_remove_fd(client->e->fd);
+    client->e->opset->close(client->e->fd, &mask, client->e);
+    eventer_free(client->e);
+    if(client->did_handshake)
+      wslay_event_context_free(client->wslay_ctx);
+    free((void *)client->path);
+    free((void *)client->service);
+    free((void *)client->host);
+    client->closed = mtev_true;
+  }
```
Comment (postwait) [inline src/mtev_websocket_client.c]: I'd add a callback here.  So the programmer can react.

**Comment by rileyberton** [comment]
Looks good to me

**Pull request merged**